### PR TITLE
Fix cue dirtiness

### PR DESCRIPTION
**Pull request by xeruf**
This fixes a regression introduced by #3016 which I am baffled stayed unnoticed for months:

Since new cues are now not assigned a track, they are never marked dirty and thus not saved. I haven't tested it extensively, but it seems that before this patch no new cues were ever saved.

It was fixed by marking cues as dirty upon initialization. It also makes the debug assertion more graceful.

Now that I write this, there would be an alternative solution: Do not mark new cues as dirty, but simply always save them when they don't have a valid id. Why that extra roundtrip?

### CODE REVIEW

**Comment by xerus2000** [comment]
Alternative solution implemented in 1c18d96ac5 - let me know whether I should revert it, was just an idea.

File: src/library/dao/cuedao.cpp [hunk 1]
```diff
@@ -208,10 +208,8 @@ void CueDAO::saveTrackCues(
     QStringList cueIds;
     cueIds.reserve(cueList.size());
     for (const auto& pCue : cueList) {
-        // New cues (without an id) must always be marked as dirty
-        DEBUG_ASSERT(pCue->getId().isValid() || pCue->isDirty());
         // Update or save cue
-        if (pCue->isDirty()) {
+        if (!pCue->getId().isValid() || pCue->isDirty()) {
```
Comment (uklotzde) [inline src/library/dao/cuedao.cpp:212]: The removed debug assertion was failing very rarely (1 or 2 times) but I was not able to reproduce it nor have any idea what could have caused it.

@Holzhaus Any insights?

File: src/library/dao/cuedao.cpp [hunk 2]
```diff
@@ -221,18 +219,16 @@ void CueDAO::saveTrackCues(
         cueIds.append(pCue->getId().toString());
     }
 
-    // Delete orphaned cues
+    // Purge orphaned cues
```
Comment (uklotzde) [inline src/library/dao/cuedao.cpp:222]: Please avoid to change code, formatting, or comments based on personal preferences without improving anything. This just causes noise.

File: src/test/cue_test.cpp [hunk 3]
```diff
@@ -10,12 +10,8 @@ namespace mixxx {
 
 TEST(CueTest, DefaultCueInfoToCueRoundtrip) {
     const CueInfo cueInfo1;
-    const Cue cueObject(
-            cueInfo1,
-            audio::SampleRate(44100),
-            true);
-    auto cueInfo2 = cueObject.getCueInfo(
-            audio::SampleRate(44100));
+    const Cue cueObject(cueInfo1, audio::SampleRate(44100));
```
Comment (uklotzde) [inline src/test/cue_test.cpp:13]: Why did you change this formatting manually?? The arguments have been placed onto single lines intentionally and our code formatting config preserve this style.

Comment (uklotzde) [reply src/test/cue_test.cpp:13]: If this is caused by your IDE then you are responsible to prevent it doing this.

Comment (Holzhaus) [reply src/test/cue_test.cpp:13]: Or just use `git add -p` and only stage the  hunks for commit that you actually want to change.

File: src/library/dao/cuedao.cpp (hunk 1 above)
Comment (xerus2000) [reply src/library/dao/cuedao.cpp:212]: I was able to trigger it reliably, that was the exactly why I made this PR ^^ but as said, I can also revert the last commit for the alternative solution.

File: src/library/dao/cuedao.cpp (hunk 2 above)
Comment (xerus2000) [reply src/library/dao/cuedao.cpp:222]: I am pretty sure we use the term "purge" whenever something gets removed irrevocably, whereas "delete" is a more broad term. So it seemed sensible to make it more precise here.

File: src/test/cue_test.cpp (hunk 3 above)
Comment (xerus2000) [reply src/test/cue_test.cpp:13]: @Holzhaus you can stop suggesting that, I almost always look at every single change before committing.

I thought that now with less arguments it might be sensible to compress this, especially as the arguments are not very interesting on their own.

File: src/library/dao/cuedao.cpp (hunk 2 above)
Comment (uklotzde) [reply src/library/dao/cuedao.cpp:222]: Purge is used in the UI. It doesn't make sense to rephrase the following SQL "DELETE"  command in the comment to "purge". Why should we use 2 different verbs in the same context??

Comment (xerus2000) [reply src/library/dao/cuedao.cpp:222]: The SQL command is rather general - for machine-readabilty. Aren't the comments supposed to be as specific and clear as possible (without sacrificing conciseness)?

Comment (uklotzde) [reply src/library/dao/cuedao.cpp:222]: I don't see any value and am not going to discuss this again.

**Comment by uklotzde** [comment]
#3599 should fix the actual cause. I suggest to close this PR.

**Comment by uklotzde** [comment]
Obsolete

**Comment by xerus2000** [comment]
@uklotzde so you basically merged what I did here: https://github.com/mixxxdj/mixxx/pull/3544/commits/04931f78d18c55873f0efe3bb81e8885384475bc

Except that I deliberately also removed the choice to initialize a cue non-dirty, since this is not a valid thing to do
